pyaedt/emit.py

```diff
@@ -117,6 +196,266 @@ def __init__(
             port=port,
             aedt_process_id=aedt_process_id,
         )
-
+        #if(self._aedt_version not in supported_versions):
+        #   raise ValueError("This version of AEDT is unsupported for PYAEDT.")
+    @pyaedt_function_handler() 
     def __enter__(self):
         return self
+    @pyaedt_function_handler()
+    def analyze(self):
+        """
+        Analyze the Active Design.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        :class:`pyaedt.modules.Revision`
+            Revision object.
+
+        References
+        ----------
+        >>> aedtapp.analyze()
+
+        """
+        design = self.odesktop.GetActiveProject().GetActiveDesign()
+        if(not self.curr_design == design.getRevision()):
+            design.AddResult()
+            self.location = self.oproject.GetPath()
+            self.revisions_list.append(Revision(self))
+            self.curr_design = design.getRevision()
+            print("checkpoint - revision generated successfully")
+            return self.revisions_list[-1]
+
+    @pyaedt_function_handler()
+    def load_revision(self, path) :
+        """
+        Load a specific revision.
+
+        Parameters
+        ----------
+        path:
+            Path to a .aedtresult file
+
+        Returns
+        -------
+        :class:`NoneType`
+
+        References
+        ----------
+        >>> aedtapp.load_revision(path)
+
+        """
+        self._emit_api.load_result(path)
+        self._result_loaded = True
+    def result_mode_error():
+        """ prints error message"""
+        print("This function is inaccessible when the Emit object has no revisions")
+    @pyaedt_function_handler()
+    def interaction_domain():
+        """
+        Return a generic interaction domain.
+
+        Parameters
+        ----------
+       
+        Returns
+        -------
+        :class:`InteractionDomain`
+
+        References
+        ----------
+        >>> Emit.interaction_domain()
+        """
+        interaction_domain = EmitApiPython.InteractionDomain()
+        return interaction_domain
+    @pyaedt_function_handler()
+    def result_type(mode =''): 
```

Comment:
Can we bind result_type to  EmitApiPython.result_type directly and avoid dealing with strings?

Reply:
Sounds good!